### PR TITLE
feat: Add mcp developer docs

### PR DESCRIPTION
**Pull request by anusshukla**
Related: AAP-53065

### CODE REVIEW

**Review by ssbarnea — APPROVED** [review]

File: docs/mcp/development.md [hunk 1]
```diff
@@ -0,0 +1,602 @@
...
+
+---
+
+## Code Style
+
+- Follow existing code style
+- Use TypeScript strict mode
+- Add JSDoc comments for public functions
+- Keep functions focused and modular
+
```
Comment (ssbarnea) [inline docs/mcp/development.md:450]: - [ ] Where are these originating from because they seem to out of sync with the reality (AI generated w/o right context?).
- [ ] There is no reason to have different contributing instructions for MCP than for the entire repository. Do not duplicate instructions, one file should be enough to cover all project from this repository 
- [ ] Out official way to run tests is `task` command, not npm!
- [ ] if we have some special mcp related contribution recommandations they can be included in a section on the general contribution docs

Comment (anusshukla) [reply docs/mcp/development.md:450]: Yes took help from cursor. Removed duplicate/generic content, switched to task mcp:test, and added MCP section to the main contributing docs.

File: docs/development/mcp.md [hunk 2]
```diff
@@ -0,0 +1,443 @@
...
+Update `src/server.ts`:
+
+#### Import handler
+
+```typescript
+import { createMyNewToolHandler } from "./handlers.js";
+```
+
+#### Register using `registerToolWithDeps`
+
```
Comment (alisonlhart) [inline docs/development/mcp.md:102]: I would expand here on how useCases, keywords, and descriptions here are important for inferring if a prompt is asking for a specific tool without naming it directly.

Make sure it's clear that these should be detailed.

File: docs/mcp/development.md [hunk 3]
```diff
@@ -0,0 +1,443 @@
...
+- [ ] Tool appears in `list_available_tools`
+- [ ] Executes correctly with valid input
+- [ ] Returns `isError` for invalid input
+- [ ] Correct JSON-RPC response
+- [ ] Clean, readable output
+- [ ] Handles edge cases
+
+---
+
+## Adding Resources
```
Comment (alisonlhart) [inline docs/mcp/development.md]: Can we put the testing section at the end, so switch resources <-> test sections?

File: docs/development/mcp.md [hunk 4]
```diff
@@ -0,0 +1,472 @@
...
+      keywords: ["keyword1", "keyword2"],
+      useCases: ["Use case 1", "Use case 2"],
+    },
+  },
+  createMyNewToolHandler(),
+  [], // dependencies (optional)
+);
+```
+
+**Important: Tool Metadata for AI Inference**
```
Comment (alisonlhart) [inline docs/development/mcp.md]:
```suggestion
#### Important: Tool Metadata for AI Inference
```
Should fix the lint problem
